### ISSUE TITLE
Repo.git.refs should support separate .git directories

### ISSUE THREAD

**Issue by joliss** [mojombo/grit#114]
```
$ git init test --separate-git-dir=test.git
Initialized empty Git repository in /home/ubuntu/tmp/test.git/
$ cat test/.git
gitdir: /home/ubuntu/tmp/test.git
$ ruby -e 'require "grit"; Grit::Repo.new("test").git.refs({}, "refs")'
/home/ubuntu/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/grit-2.5.0/lib/grit/git-ruby.rb:116:in `chdir': Not a directory - /home/ubuntu/tmp/test/.git (Errno::ENOTDIR)
    from /home/ubuntu/.rbenv/versions/1.9.3-p194/lib/ruby/gems/1.9.1/gems/grit-2.5.0/lib/grit/git-ruby.rb:116:in `refs'
    from -e:1:in `<main>'
```

This "separate git dir" feature (added in git 1.7.5) is unfortunately not as obscure as it seems, since new gits will use it by default when initializing submodules.

I'm not sure how many places in grit are affected -- there seems to be a bunch of calls to chdir across the source tree. Perhaps it would be sensible to run large chunks of the test suite against both regular working copies and working copies with separate .git dirs.

**Comment by ReneB** [comment]
I think the commit above may fix this, but I didn't get around to writing tests for it yet (yeah I know, boo hiss for not doing TDD on this one). I just found out that after the umpteenth repository-breaking sed-command and finally separating the .git dir, git-up stopped working and I found out it was caused by this issue.

I do want to write tests for it but right now it was just a I-really-gotta-get-back-to-work kind of commit, HTH.
